src/crypto-providers/util.ts

```diff
@@ -267,6 +255,10 @@ const determineSigningMode = (
       : SigningMode.POOL_REGISTRATION_AS_OWNER
   }
 
+  if (txBody.collaterals || txBody.requiredSigners) {
```

Comment:
`determineSigningMode` is very fragile. At the least it requires detailed comments explaining why it is written exactly like this. For instance, one of my immediate ideas was to move the multisig signing files check above txBody checks (because it seems like an explicitly expressed intent --- which it actually is not, since we allow 1854 within Plutus).

Reply:
Good point, comments added in https://github.com/vacuumlabs/cardano-hw-cli/pull/112/commits/760d83b1949aa852608df595426094d89e8f70f1